ra-optimisation/rao-api/src/main/java/com/farao_community/farao/rao_api/Rao.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (c) 2019, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.farao_community.farao.rao_api;
+
+import com.farao_community.farao.commons.FaraoException;
+import com.farao_community.farao.data.crac_api.Crac;
+import com.farao_community.farao.ra_optimisation.RaoComputationResult;
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
+import com.powsybl.commons.Versionable;
+import com.powsybl.commons.config.PlatformConfig;
+import com.powsybl.commons.util.ServiceLoaderCache;
+import com.powsybl.computation.ComputationManager;
+import com.powsybl.computation.local.LocalComputationManager;
+import com.powsybl.iidm.network.Network;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+/**
+ * RA optimisation main API. It is a utility class (so with only static methods) used as an entry point for running
+ * a Ra optimisation allowing to choose either a specific find implementation or just to rely on default one.
+ *
+ * @author Baptiste Seguinot {@literal <baptiste.seguinot at rte-france.com>}
+ */
+public final class Rao {
+
+    private Rao() {
+        throw new AssertionError("Utility class should not been instantiated");
+    }
+
+    private static final Supplier<List<RaoProvider>> PROVIDERS_SUPPLIERS
+            = Suppliers.memoize(() -> new ServiceLoaderCache<>(RaoProvider.class).getServices());
+
+    /**
+     * A RA optimisation runner is responsible for providing convenient methods on top of {@link RaoProvider}:
+     * several variants of synchronous and asynchronous run with default parameters.
+     */
+    public static class Runner implements Versionable {
+
+        private final RaoProvider provider;
+
+        public Runner(RaoProvider provider) {
+            this.provider = Objects.requireNonNull(provider);
+        }
+
+        public CompletableFuture<RaoComputationResult> runAsync(Network network, Crac crac, String workingStateId, ComputationManager computationManager, RaoParameters parameters) {
+            Objects.requireNonNull(workingStateId);
+            Objects.requireNonNull(parameters);
```

Comment:
I think it would be useful to add a specific message "workingStateId should not be null" for each of the parameters we don't want to be null.

Reply:
I would also add Objects.requireNonNull on network and crac

Reply:
added with network and crac, with specific messages